sklearn/ensemble/_base.py

```diff
@@ -16,13 +16,13 @@
 
 
 def _fit_single_estimator(
-    estimator, X, y, sample_weight=None, message_clsname=None, message=None
+    estimator, X, y, fit_params, sample_weight=None, message_clsname=None, message=None
 ):
     """Private function used to fit an estimator within a job."""
     if sample_weight is not None:
```

Comment:
this logic of checking for sample_weight only exists because we didn't have metadata routing so far. So whenever metadata routing is enabled, we just pass `fit_params` merged with sample_weight. So it would be sth like:

```py
    # TODO(SLEP6): remove sample_weight as an argument when metadata routing is
    # always enabled.
    if _routing_enabled():
        if sample_weight is not None:
            fit_params['sample_weight'] = sample_weight
        with _print_elapsed_time(message_clsname, message):
            return estimator.fit(X, y, **fit_params)
        
    if sample_weight is not None:
        try:
            with _print_elapsed_time(message_clsname, message):
                estimator.fit(X, y, sample_weight=sample_weight)
        except TypeError as exc:
            if "unexpected keyword argument 'sample_weight'" in str(exc):
                raise TypeError(
                    "Underlying estimator {} does not support sample weights.".format(
                        estimator.__class__.__name__
                    )
                ) from exc
            raise
    else:
        with _print_elapsed_time(message_clsname, message):
            estimator.fit(X, y)
    return estimator
```